adapters/andalucia.js

```diff
@@ -0,0 +1,920 @@
+      }
+      res = res[0].concat(res[1]);
+      done(null, res);
+    });
+  };
+};
+
+const formatData = ($, stations) => {
+  const tables = [];
+  const stationNames = stations.map((station) => { return station.name; });
+  // each station data sits in consecutive tables
+  // the first table has the station name, the second the actual aq data.
+  // the below logic checks to see if a table has a station name
+  // and if it does, pushes it and the consecutive table to tables
+  $('table').each((i, el) => {
+    const stationMatch = find(stationNames, (stationName) => {
+      const tableText = $(el).text();
+      return tableText.match(stationName);
+    });
+    if (stationMatch) {
+      const stationTop = $($('table')[i]);
+      const stationBottom = $($('table')[i + 1]);
+      tables.push(stationTop);
+      tables.push(stationBottom);
+    }
+  });
+  // data is held two consecutive tables
+  // (one with station name, other actual data)
+  let stationData = chunk(tables, 2);
+  // return a list of measurement objects for each table.
+  return stationData.map((table) => {
+    const station = $(table[0]['0'].children[2]).text().split('Estacion')[1].trim();
+    const stationLoc = mapStationCoords(station);
+    table = makeTable(table[1]['0'], $);
+    return makeMeasurements(table, stationLoc, station);
+  });
+};
+
+const mapStationCoords = (station) => {
+  return stations.filter((stationObj) => {
+    return stationObj['name'] === station;
+  }).map((stationObj) => {
+    return {
+      latitude: stationObj['coordinates'][1],
+      longitude: stationObj['coordinates'][0]
+    };
+  })[0];
+};
+
+const makeTable = (cheerioTable, $) => {
+  return cheerioTable.children.filter((child) => {
+    return child.name === 'tr';
+  }).map((row) => {
+    const tds = [];
+    $(row).children().each((i, el) => {
+      tds.push($(el).text());
+    });
+    return tds;
+  });
+};
+
+const makeMeasurements = (stationData, stationLoc, stationName) => {
+  // get index of data sources we can record
+  const pollutantIndexes = stationData[0].filter((cell) => {
+    return includes(acceptableParameters.slice(3, 5), cell.toLowerCase());
+  }).map((validCell) => {
+    return stationData[0].indexOf(validCell);
+  });
+  // include index for 1st column, which has the date
+  pollutantIndexes.push(0);
+  // filter rows for only the date,SO2, and NO2 rows
+  stationData = stationData.map((row) => {
+    return row.filter((cell, index) => {
+      if (includes(pollutantIndexes, index)) {
+        return cell;
+      }
+    });
+  });
+  // make each row into a list of measurements
+  stationData = stationData.slice(1, -1).map((row, index) => {
+    return row.map((cell, index) => {
+      if (index !== 0) {
+        // ignore blank spaces
+        if (/\S/.test(cell)) {
+          console.log(cell);
+          return {
+            city: stationName.replace(/\w\S*/g, (t) => { return t.charAt(0).toUpperCase() + t.substr(1).toLowerCase(); }),
```

Comment:
Sample output looks like 

```json
{"date":{"utc":"2017-07-27T22:00:00.000Z","local":"2017-07-28T00:00:00+02:00"},"parameter":"no2","location":"unused","value":9,"unit":"µg/m³","city":"Cobre Las Cruces","attribution":[{"name":"Ministry of Environment and Spatial Planning","url":"http://www.juntadeandalucia.es/medioambiente/site/portalweb"}],"averagingPeriod":{"unit":"hours","value":1},"coordinates":{"latitude":37.5550190832439,"longitude":-6.03616087321919},"country":"ES","sourceName":"Andalucia","sourceType":"government","mobile":false}
```

Looks like we're missing `location`, everything should have a unique location within a `city`.